example error with typescript 2.6.1

<!--
Please make sure to provide a **minimal and self-contained reproduction** if you are reporting a bug.
You can use the following jsfiddle template or make an example github repository.
https://jsfiddle.net/ktsn/nm55jnjk/

For usage questions, please use the following resources:

* Ask on the forum https://forum.vuejs.org
* Ask on the chat room https://chat.vuejs.org
* Look for / ask questions on Stack Overflow
-->

```shell
$ npm install && npm run example
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

> vue-class-component@6.0.0 example D:\Test\vue-class-component
> npm run build && webpack --config example/webpack.config.js


> vue-class-component@6.0.0 build D:\Test\vue-class-component
> npm run build:ts && npm run build:main


> vue-class-component@6.0.0 build:ts D:\Test\vue-class-component
> tsc -p .

src/component.ts(38,16): error TS2532: Object is possibly 'undefined'.
src/component.ts(40,58): error TS2532: Object is possibly 'undefined'.
src/component.ts(41,16): error TS2532: Object is possibly 'undefined'.
src/component.ts(41,34): error TS2532: Object is possibly 'undefined'.
src/component.ts(44,14): error TS2532: Object is possibly 'undefined'.
src/component.ts(45,14): error TS2532: Object is possibly 'undefined'.

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build:ts"
npm ERR! node v6.10.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! vue-class-component@6.0.0 build:ts: `tsc -p .`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the vue-class-component@6.0.0 build:ts script 'tsc -p .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the vue-class-component package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tsc -p .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs vue-class-component
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls vue-class-component
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Test\vue-class-component\npm-debug.log

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
npm ERR! node v6.10.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! vue-class-component@6.0.0 build: `npm run build:ts && npm run build:main`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the vue-class-component@6.0.0 build script 'npm run build:ts && npm run build:main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the vue-class-component package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build:ts && npm run build:main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs vue-class-component
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls vue-class-component
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Test\vue-class-component\npm-debug.log

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "example"
npm ERR! node v6.10.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! vue-class-component@6.0.0 example: `npm run build && webpack --config example/webpack.config.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the vue-class-component@6.0.0 example script 'npm run build && webpack --config example/webpack.config.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the vue-class-component package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build && webpack --config example/webpack.config.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs vue-class-component
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls vue-class-component
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Test\vue-class-component\npm-debug.log

```

package info:

```shell
$ npm ls --depth 0
vue-class-component@6.0.0 D:\Test\vue-class-component
+-- @types/chai@4.0.4
+-- @types/mocha@2.2.44
+-- babel-core@6.26.0
+-- babel-loader@7.1.2
+-- babel-plugin-transform-class-properties@6.24.1
+-- babel-plugin-transform-decorators-legacy@1.3.4
+-- babel-preset-es2015@6.24.1
+-- chai@4.1.2
+-- css-loader@0.28.7
+-- mocha@4.0.1
+-- rimraf@2.6.2
+-- rollup@0.50.0
+-- rollup-plugin-replace@2.0.0
+-- testdouble@3.2.6
+-- ts-loader@2.3.7
+-- typescript@2.6.1
+-- uglify-js@3.1.6
+-- vue@2.5.2
+-- vue-loader@13.3.0
+-- vue-template-compiler@2.5.2
`-- webpack@3.8.1
```

--- a/src/component.ts
+++ b/src/component.ts
@@ -34,7 +34,7 @@ export function componentFactory (
       options[key] = proto[key]
       return
     }
-    const descriptor = Object.getOwnPropertyDescriptor(proto, key)
+    const descriptor = Object.getOwnPropertyDescriptor(proto, key)!
     if (typeof descriptor.value === 'function') {
       // methods
       (options.methods || (options.methods = {}))[key] = descriptor.value

